.github/workflows/push.yaml

```
@@ -0,0 +1,141 @@
+        env:
+          NEW_CHARTS: ${{ steps.detect_new_charts.outputs.new_charts }}
+          GH_TOKEN: ${{ secrets.CHARTS_REPO_PAT }}
+          COMMIT_SHA: ${{ github.sha }}
+        run: |
+          set -eux
+
+          if [ -z "$NEW_CHARTS" ]; then
+            echo "No new charts detected. Exiting."
+            exit 0
+          fi
+
+          # --- Dependency Installation ---
+          ARCH=$(uname -m)
+          if [ "$ARCH" = "x86_64" ]; then SYSTEM_ARCH="amd64"; elif [ "$ARCH" = "aarch64" ]; then SYSTEM_ARCH="arm64"; else echo "Unsupported architecture: $ARCH"; exit 1; fi
+          curl -L "https://github.com/mikefarah/yq/releases/latest/download/yq_linux_${SYSTEM_ARCH}" -o /usr/local/bin/yq && chmod +x /usr/local/bin/yq
+          curl -L "https://github.com/cli/cli/releases/download/v2.82.0/gh_2.82.0_linux_${SYSTEM_ARCH}.tar.gz" -o gh.tar.gz && tar -zxvf gh.tar.gz && mv "gh_2.82.0_linux_${SYSTEM_ARCH}/bin/gh" /usr/local/bin/gh && rm -rf gh.tar.gz "gh_2.82.0_linux_${SYSTEM_ARCH}"
+
+          # --- Download Freeze Manifest ---
+          curl -L "https://raw.githubusercontent.com/rancher/org/refs/heads/main/manifests/resources/RepositoryRuleset/rancher/code-freeze.yaml?token=${GH_TOKEN}" -o /tmp/code-freeze.yaml
+
+          # --- Group charts by target branch ---
+          rm -rf /tmp/branch_data && mkdir /tmp/branch_data
+          for chart_full_version in $NEW_CHARTS; do
+            CHART_NAME=$(echo $chart_full_version | cut -d'/' -f1)
+            CHART_VERSION=$(echo $chart_full_version | cut -d'/' -f2)
+            UPSTREAM_VERSION=$(echo $CHART_VERSION | cut -d'.' -f1,2)
+          
+            yq e ".packages.$CHART_NAME.\"$UPSTREAM_VERSION\".branches" -o=json "charts-config.yaml" | jq -c '.[]' | while read -r branch_config; do
+              TARGET_BRANCH=$(echo "$branch_config" | jq -r '.branch')
+              PACKAGE_PATH=$(echo "$branch_config" | jq -r '.package')
+              echo "${chart_full_version},${PACKAGE_PATH}" >> "/tmp/branch_data/${TARGET_BRANCH}"
+            done
+          done
+
+          # --- Checkout downstream repo ---
+          git clone https://oauth2:${GH_TOKEN}@github.com/rancher/charts.git charts-repo
+          cd charts-repo
+          git config user.name "github-actions[bot]"
+          git config user.email "github-actions[bot]@users.noreply.github.com"
+
+          # --- Process each target branch ---
+          for branch_file in /tmp/branch_data/*; do
+            TARGET_BRANCH=$(basename "$branch_file")
+
+            # --- Check if branch is frozen ---
+            MANIFEST_BRANCH_NAME=$(echo "$TARGET_BRANCH" | sed 's|dev-v|release/v|')
+            MANIFEST_REF="refs/heads/$MANIFEST_BRANCH_NAME"
+            IS_FROZEN=$(yq e ".spec.forProvider.conditions[].refName[].include[] | select(. == \"$MANIFEST_REF\")" /tmp/code-freeze.yaml)
+            if [ -n "$IS_FROZEN" ]; then
+              echo "Branch $MANIFEST_BRANCH_NAME is frozen. Skipping."
```

```suggestion
              echo "Branch $MANIFEST_BRANCH_NAME is frozen. Skipping." | tee -a "$GITHUB_STEP_SUMMARY"
```

The [summary](https://github.blog/news-insights/product-news/supercharging-github-actions-with-job-summaries/#create-summaries) feature is a cool way to make output of GHA runs have more details. I think if we use `tee` here we can have this printed to the stdout the job runs in and set as the Summary.